### PR TITLE
feat(selinux): Get rid of all file WX of Trivalent's Selinux `trivalent_domain`

### PR DESCRIPTION
**Pull request by PhysicsIsAwesome**
This gets rid of all file WX of Trivalent's Selinux `trivalent_domain` I could find:

- cache_home_t
- chrome_sandbox_home_t
- config_home_t
- trivalent_home_t
- trivalent_domain itself

It introduces a new on-by-default tunable `trivalent_exec_flatpaks` so people can disable execution of Flatpaks. Execution is needed for certain extensions being able to talk to their app counterpart like Keepassxc. 

Additionally there is a tunable `trivalent_drm` which is on by-default. This allows file WX and needs to be turned off by users who want W^X at the expense of not being able to use DRM restricted content. 

Testing done:

- Started Trivalent
- Visited a few websites including video content (Youtube)
- Checked AVC logs
- Deleted trivalent's folder in `.config` 

Might need additional testing for edge cases

### CODE REVIEW

File: files/scripts/selinux/trivalent/trivalent.fc [hunk 1]
```diff
@@ -9,4 +9,5 @@
 /usr/lib/trivalent/trivalent.sh				--	gen_context(system_u:object_r:trivalent_script_exec_t,s0)
 /usr/lib/trivalent/install_filter.sh				--	gen_context(system_u:object_r:trivalent_script_exec_t,s0)
 HOME_DIR/\.cache/trivalent(/.*)?		gen_context(system_u:object_r:trivalent_home_t,s0)
-HOME_DIR/\.config/trivalent(/.*)?	gen_context(system_u:object_r:trivalent_home_t,s0)
\ No newline at end of file
+HOME_DIR/\.config/trivalent(/.*)?	gen_context(system_u:object_r:trivalent_home_t,s0)
+HOME_DIR/\.config/trivalent(/.*)?\.so           --  gen_context(system_u:object_r:trivalent_home_lib_t,s0)
```
Comment (RoyalOughtness) [inline files/scripts/selinux/trivalent/trivalent.fc]: how will this fcontext be updated?

Comment (PhysicsIsAwesome) [reply files/scripts/selinux/trivalent/trivalent.fc]: Do you mean during installation time or typetransition Selinux rule?

Comment (RoyalOughtness) [reply files/scripts/selinux/trivalent/trivalent.fc]: @PhysicsIsAwesome no i mean how will the selinux file contexts be modified, since these are in the homedir. Image upgrades don't touch the homedir

Comment (PhysicsIsAwesome) [reply files/scripts/selinux/trivalent/trivalent.fc]: Good catch. Seems like Trivalent can create `~/.config/trivalent` on-the-fly, so it can recover from users deleting it or new Linux user creation. The only `*.so` file I found, is a widevine file. This also can get created on-the-fly and even gets created when visiting some website, if DRM is deactivated in Trivalent. I will add a type_transition, for the file contexts to properly apply, but this also requires create and write permissions. For DRM to be usable, it seems to need mmap(PROT_EXEC) on that file. Since we probably don't want to break DRM for all users, but also want users to give the option to not have file WX, I would put this behind a tunable. A few design questions arise:
- Default behavior: Widevine/DRM allowed or disallowed?
- Include only the mmap_execute in this tunable or also the manage permission?
- In case of DRM tunable set to off, shall it be silenced?

Comment (RoyalOughtness) [reply files/scripts/selinux/trivalent/trivalent.fc]: Widevine should work by default, and probably only the execute because it should still be able to be written and updated even if it can't be executed.

Comment (PhysicsIsAwesome) [reply files/scripts/selinux/trivalent/trivalent.fc]: I added handling of DRM via boolean, type_transition and managing the corresponding file type. It still needs a one time restorecon for file pattern `HOME_DIR/\.config/trivalent/WidevineCdm(/.*)?\.so` for existing users. How is this usually handled in secureblue?

Comment (HastD) [reply files/scripts/selinux/trivalent/trivalent.fc]: You could add a [tmpfiles.d](https://www.freedesktop.org/software/systemd/man/latest/tmpfiles.d.html) configuration that does this. What this would look like is a file in `/usr/share/user-tmpfiles.d` called something like `trivalent-home-fix-secontexts.conf` with the following line:
```
Z %h/.config/trivalent
```
This is effectively the same as each user running `restorecon -R ~/.config/trivalent` on boot. It's not a *one-time* restorecon, but we could simply remove this configuration after the next ISO build or something. (The alternative would be to write a service that does the restorecon manually and then creates a stamp file that prevents it from running more than once, but the tmpfiles.d approach is simpler and easier to maintain.)

**Comment by HastD** [comment]
Just a heads up, we're in the process of moving Trivalent's SELinux policy to [the Trivalent repo](https://github.com/secureblue/Trivalent) (in the `build` directory), where it will be built as a `trivalent-selinux` RPM subpackage. This will allow the SELinux policy to be installed alongside Trivalent on any RPM-based distro.

So, it would probably be best to open a PR with these changes in the Trivalent repo rather than here.

**Comment by HastD** [comment]
Closing this PR since Trivalent's SELinux policy has now been moved to the Trivalent repo as mentioned above.
